### DISCUSSION TITLE
Disable Sorting on Header Click

### DISCUSSION THREAD

**Discussion by applegirl21** [KevinVandy/material-react-table#383] [Q&A] [answered]
Is it possible to disable sorting on header click, but still keep the sort buttons? I checked the docs and didn’t see anything. There is a [discussion](https://github.com/KevinVandy/material-react-table/discussions/198) about disabling the third sort click on the header, but the solution didn’t work. When I tried to add the enableSortingRemoval: false to the column definition, it said the property didn’t exist. I was able to add it on the table itself, but it still didn’t work.

**Comment by KevinVandy** [comment, marked as answer]
[`enableSortingRemoval`](https://tanstack.com/table/v8/docs/api/features/sorting#enablesortingremoval) is for just whether or not toggling sorting includes the 3rd click click to unsort. And no I don't think it is currently possible to disable the current click behavior. But it is made this way for good accessibility. I would think twice before removing it.

**Reply by applegirl21** [reply]
Is it possible to add custom buttons to the headers without triggering the sort button? I used the Header option, but it causes the sort button to be toggled whenever my custom button is clicked because it’s part of the header.

**Reply by MarkusEllyton** [reply]
You could stop the propagation of the click event

```typescript
Header: () => {
            return <button onClick={(e) => {e.stopPropagation(); console.log("Clicked!")}}>Click me!</button>
        }
```
